From 83a5ebf57952458f3d2f9d18a8742e9e1679124d Mon Sep 17 00:00:00 2001
From: shubang <shubang@google.com>
Date: Fri, 13 Aug 2021 15:42:15 -0700
Subject: [PATCH 61/66] [Q_asb_2021-11] TIF: fix issue of using caller-aware
 methods after clearCallingIdentity()

Bug: 189824175
Test: atest android.media.tv.cts.TvInputManagerTest
Change-Id: Iced6c2245c0099bc4bcdaceb51f8cce4dbc0a392
(cherry picked from commit a2ab2772d70984b1c24d5d4509b9c7a5abea71f9)
(cherry picked from commit 05363ef193ae231fc7a1fe666de88861e1084b17)
---
 .../java/com/android/server/tv/TvInputManagerService.java    | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/services/core/java/com/android/server/tv/TvInputManagerService.java b/services/core/java/com/android/server/tv/TvInputManagerService.java
index 057b53e6232f..20a6cf4f60f6 100644
--- a/services/core/java/com/android/server/tv/TvInputManagerService.java
+++ b/services/core/java/com/android/server/tv/TvInputManagerService.java
@@ -1981,10 +1981,9 @@ public final class TvInputManagerService extends SystemService {
         public void requestChannelBrowsable(Uri channelUri, int userId)
                 throws RemoteException {
             final String callingPackageName = getCallingPackageName();
+            final int resolvedUserId = resolveCallingUserId(Binder.getCallingPid(),
+                    Binder.getCallingUid(), userId, "requestChannelBrowsable");
             final long identity = Binder.clearCallingIdentity();
-            final int callingUid = Binder.getCallingUid();
-            final int resolvedUserId = resolveCallingUserId(Binder.getCallingPid(), callingUid,
-                userId, "requestChannelBrowsable");
             try {
                 Intent intent = new Intent(TvContract.ACTION_CHANNEL_BROWSABLE_REQUESTED);
                 List<ResolveInfo> list = getContext().getPackageManager()
-- 
2.34.1

